model/src/lib.rs

```diff
@@ -37,6 +38,24 @@ fn hashed_eip712_message(domain_separator: &DomainSeparator, struct_hash: &[u8;
     signing::keccak256(&message)
 }
 
+fn hashed_ethsign_message(domain_separator: &DomainSeparator, struct_hash: &[u8; 32]) -> [u8; 32] {
+    let mut message = [0u8; 60];
+    message[..28].copy_from_slice(b"\x19Ethereum Signed Message:\n32");
+    message[28..].copy_from_slice(&hashed_eip712_message(domain_separator, struct_hash));
+    signing::keccak256(&message)
+}
+
+fn signing_message(
```

Comment:
naming nit: to be more inline with the ethsign/eip712 methods maybe `hashed_signing_message()` (I kind of like the `digest` word e.g. `ethsign_message_digest` in the previous version).